### PR TITLE
New rule: disallowUnusedParams

### PR DESCRIPTION
**Pull request by markelog**
Ref #1771

### CODE REVIEW

**Comment by hzoo** [comment]
Interesting so we're adding escope

File: lib/token-assert.js [hunk 1]
```diff
@@ -570,7 +570,7 @@ TokenAssert.prototype.noTrailingSpaces = function(options) {
                 }
             }
 
-            if (typeof targetToken !== 'undefined' && !fixed) {
+            if (typeof targetToken !== null && !fixed) {
```
Comment (mikesherov) [inline lib/token-assert.js]: missed one.

**Comment by mikesherov** [comment]
If we're adding this, also remove `unused` from `jshintrc` please

**Comment by markelog** [comment]
> If we're adding this

We discussed this couple times in gitter, it seems we were generally liking the idea. Plus it is pretty useful to have is as autofix.

> also remove unused from jshintrc please

`unused` also catches the vars, whereas this rule works only for params

**Pull request closed without merging**